google/gax/__init__.py

```diff
@@ -482,3 +489,160 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    if not any_pb.Is(pb_type.DESCRIPTOR):
+        raise TypeError(
+            'Could not convert {} to {}'.format(
+                any_pb.__class__.__name__, pb_type.__name__))
+
+    return pb_type.FromString(any_pb.value)
+
+
+class ResultError(GaxError):
```

Comment:
Right. The only `GaxError` subclass we've defined so far is `RetryError`, and that's only because it pertains to a specific client-side feature that we implement (automatic retry for transient errors on idempotent methods). Errors associated with LRO will be more along the lines of server-side errors we get from synchronous methods, which we just wrap in plain `GaxError`.